### PR TITLE
HomeWork: Full house (Part1-3) (Aleksandr_Saunin@epam.com)

### PR DESCRIPTION
**Pull request by asaunin**

### CODE REVIEW

File: src/test/java/part2/exercise/CollectorsExercise2.java [hunk 1]
```diff
@@ -163,95 +163,105 @@ public MapPair(Map<String, Key> keyById, Map<String, List<Value>> valueById) {
...
+        @Override
+        public Supplier<MapPair> supplier() {
+            return MapPair::new;
+        }
+
+        @Override
+        public BiConsumer<MapPair, Pair> accumulator() {
+            return (m, p) -> {
+                final Key key = p.getKey();
+                final String sKey = key.toString();
```
Comment (senia-psm) [inline src/test/java/part2/exercise/CollectorsExercise2.java]: Don't use `toString`. Use `getId`.

File: src/test/java/part2/exercise/CollectorsExercise2.java [hunk 2]
```diff
@@ -163,95 +163,105 @@ public MapPair(Map<String, Key> keyById, Map<String, List<Value>> valueById) {
...
+                final Value value = p.getValue();
+                final Map<String, Key> keyMap = m.getKeyById();
+                final Map<String, List<Value>> valueMap = m.getValueById();
+                keyMap.put(sKey, key);
+                if (!valueMap.containsKey(sKey))
+                    valueMap.put(sKey, new ArrayList<>());
+                valueMap.get(sKey).add(value);
```
Comment (senia-psm) [inline src/test/java/part2/exercise/CollectorsExercise2.java]: `Key` and `Value` in `Pair` are not related. You should use `Key::getId` and `Value::getKeyId`

**Review by senia-psm — CHANGES_REQUESTED** [review]

File: src/test/java/part2/exercise/CollectorsExercise2.java (hunk 2 above)
Comment (ASaunin) [inline src/test/java/part2/exercise/CollectorsExercise2.java]: In our task this point was not mentioned. I realized it too late, while doing the last test in CollecrorsCombiner module...
Whether is it necessary to improve all the modules in code, where my realization had take place?

File: src/test/java/part2/exercise/CollectorsExercise2.java [hunk 3]
```diff
@@ -163,95 +163,105 @@ public MapPair(Map<String, Key> keyById, Map<String, List<Value>> valueById) {
...
-        // В два прохода
-        // final Map<String, Key> keyMap1 = pairs.stream()...
-
-        // final Map<String, List<Value>> valuesMap1 = pairs.stream()...
-
-        // В каждом Map.Entry id ключа должно совпадать с keyId для каждого значения в списке
-        // final Map<Key, List<Value>> keyValuesMap1 = valueMap1.entrySet().stream()...
-
-        // В 1 проход в 2 Map с использованием MapPair и mapMerger
-        final MapPair res2 = pairs.stream()
```
Comment (ASaunin) [inline src/test/java/part2/exercise/CollectorsExercise2.java]: There is MapPairCollector class and it's implementation in function collectKeyValueMapPair

**Pull request closed without merging**